src/crimson/os/seastore/journal.cc

```diff
@@ -147,6 +147,13 @@ ceph::bufferlist Journal::encode_record(
     sizeof(checksum_t),
     reinterpret_cast<const char *>(&metadata_crc_le));
 
+  bufferptr bp(ceph::buffer::create_aligned(bl.length(), block_size));
+  bp.zero();
+  auto iter = bl.cbegin();
+  iter.copy(bl.length(), bp.c_str());
+  bl.clear();
+  bl.append(bp);
```

Comment:
i think this is equivalent to `bl.rebuild()`, am i right?

Reply:
That's because data sent to the underlying blk layer has to be aligned to the block size

Reply:
I mean each iovec has to be aligned to the block size

Reply:
Oh, sorry, I misunderstood you, will correct it.